tools/test-js-with-node

```diff
@@ -184,6 +184,7 @@ EXEMPT_FILES = make_set(
         "static/js/timerender.ts",
         "static/js/tippyjs.js",
         "static/js/todo_widget.js",
+        "static/js/topic_generator.js",
```

Comment:
One option to include a test would be to add a parameter to next_topic(). This parameter would be a function that encapsulates lines 22-25 in this pull request and would return an array of the topics that are filtered by the left-sidebar. 

This would allow the test suite to override the filter function to return a hardcoded array of topics that fit a filter. With permission to add this parameter to next_topic(), I can add such a test to the test suite to maintain the coverage requirement.

Reply:
One could probably instead just set up the `topic_list_data` data structure, right?  Generally we prefer that over this sort of mocking approach, since it's more of an end-to-end test.